src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1096,5 +1102,36 @@ private static void ThrowMinMaxException<T>(T min, T max)
         {
             throw new ArgumentException(SR.Format(SR.Argument_MinMaxValue, min, max));
         }
+
+        public static double ScaleB(double f, int n)
+        {
+            double y = f;
+            if (n > 1023)
+            {
+                y *= SCALEB_C1;
+                n -= 1023;
+                if (n > 1023)
+                {
+                    y *= SCALEB_C1;
+                    n -= 1023;
+                    if (n > 1023)
+                        n = 1023;
```

Comment:
style nit: surround with brackets to match the existing style - elsewhere in this file, curly brackets around single line conditions are used https://github.com/dotnet/runtime/blob/57c49318b4de61a17cce116c3d9a2dee3cd4f31a/src/libraries/System.Private.CoreLib/src/System/Math.cs#L301-L318